### PR TITLE
RHOAIENG-6517 Automate support for configuration of controller resources

### PR DESCRIPTION
**Pull request by asanzgom**
This Test Case validates that is possible to edit the resources fields inside controller deployment Yaml files for KServe and Model Mesh and the new values do not get overwritten by the Operator except for the fields outside of resources.

### CODE REVIEW

**Comment by asanzgom** [comment]
rhods-ci-pr-test/2959 ran successfully

**Review by FedeAlonso — CHANGES_REQUESTED** [review]

File: ods_ci/tests/Tests/100__deploy/130__operators/130__rhods_operator/142__dsc_components.robot [hunk 1]
```diff
@@ -136,6 +137,35 @@ Validate Datasciencepipelines Removed State
...
+        Should Be Equal As Integers    ${rc}    ${0}
+        ${rc}    ${out}=    Run And Return Rc And Output
+        ...    oc patch Deployment ${controller} -n ${APPLICATIONS_NAMESPACE} --type=json -p="[{'op': 'replace', 'path': '/spec/template/spec/serviceAccountName', 'value': 'random-sa-name'}]"    # robocop: disable
+        Should Be Equal As Integers    ${rc}    ${0}
+        Sleep  40  reason=Wait until Deployment is reloaded after being overwritten by Operator
+        @{d_obj} =  OpenShiftLibrary.Oc Get  kind=Deployment  name=${controller}    namespace=${APPLICATIONS_NAMESPACE}
+        &{d_obj_dictionary} =  Set Variable  ${d_obj}[0]    
+        ${cpu_limit}=    Set Variable    ${d_obj_dictionary.spec.template.spec.containers[0].resources.limits.cpu}
+        ${memory_limit}=    Set Variable    ${d_obj_dictionary.spec.template.spec.containers[0].resources.limits.memory}
+        ${sa_name}=    Set Variable    ${d_obj_dictionary.spec.template.spec.serviceAccountName}
```
[the remark this answers is not in the document]
Comment (asanzgom) [reply ods_ci/tests/Tests/100__deploy/130__operators/130__rhods_operator/142__dsc_components.robot:158]: @FedeAlonso Can you explain this requested change? Is it formatting?

[the remark this answers is not in the document]
Comment (FedeAlonso) [reply ods_ci/tests/Tests/100__deploy/130__operators/130__rhods_operator/142__dsc_components.robot:158]: No, it's just that that variable is not used (maybe I'm wrong)

**Review by FedeAlonso — APPROVED** [review]

File: ods_ci/tests/Tests/100__deploy/130__operators/130__rhods_operator/142__dsc_components.robot (hunk 1 above)
[the remark this answers is not in the document]
Comment (asanzgom) [reply ods_ci/tests/Tests/100__deploy/130__operators/130__rhods_operator/142__dsc_components.robot]: Removed

**Review by CFSNM — APPROVED** [review]
LGTM

**Review by FedeAlonso — APPROVED** [review]

**Pull request merged**